src/renderer/stores/locations.js

```diff
@@ -13,13 +14,9 @@ const locationStore = new Store(defaultState)
 
 const getLocations = (chatId, mapSettings) => {
   const { timestampFrom, timestampTo } = mapSettings
-  ipcRenderer.send(
-    'EVENT_DC_FUNCTION_CALL',
+  callDcMethod(
```

Comment:
Can you refactor those methods to also return the locations? Otherwise we still need to listen for another event to get back the results. (If that's possible without too much refactoring)